### PR TITLE
Run3-hcx320 Use of ESGetToken in RecoLocalCalo/HcalRecAlgos (replacing #35234)

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Use of ESGetToken in RecoLocalCalo/HcalRecAlgos (replacing #35234)

#### PR validation:

Use the runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc [hunk 1]
```diff
@@ -88,14 +94,11 @@ void HcalSevLvlAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup
 #endif
 
 #ifdef THIS_IS_AN_EVENTSETUP_EXAMPLE
-  ESHandle<SetupData> pSetup;
-  iSetup.get<SetupRecord>().get(pSetup);
+  auto pSetup = &iSetup.getData(tokSet_);
 #endif
```
Comment (tvami) [inline RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc:98]: Now that the event setup happens with the `getData` method, I believe these lines are not needed anymore

File: RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc [hunk 2]
```diff
@@ -62,8 +65,11 @@ class HcalSevLvlAnalyzer : public edm::EDAnalyzer {
 // constructors and destructor
 //
 HcalSevLvlAnalyzer::HcalSevLvlAnalyzer(const edm::ParameterSet& iConfig)
-
-{
+    :
+#ifdef THIS_IS_AN_EVENTSETUP_EXAMPLE
+      totkSet_(esConsumes()),
+#endif
```
Comment (tvami) [inline RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc:71]: Now that the event setup happens with the `getData` method, I believe these lines are not needed anymore

File: RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc [hunk 3]
```diff
@@ -37,17 +36,21 @@
...
-  virtual void analyze(const edm::Event&, const edm::EventSetup&);
-  virtual void endJob();
+  void beginJob() override;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
 
   // ----------member data ---------------------------
+#ifdef THIS_IS_AN_EVENTSETUP_EXAMPLE
+  const edm::ESGetToken<SetupData, SetupRecord> totkSet_;
+#endif
```
Comment (tvami) [inline RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc:52]: Now that the event setup happens with the `getData` method, I believe these lines are not needed anymore

File: RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc]: removed

File: RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc (hunk 2 above)
Comment (bsunanda) [reply RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc]: removed

File: RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc (hunk 3 above)
Comment (bsunanda) [reply RecoLocalCalo/HcalRecAlgos/test/HcalSevLvlAnalyzer.cc]: Removed

**Comment by abdoulline** [comment]
For the attention of  @igv4321

**Comment by jpata** [comment]
+reconstruction
- for cfaf1ba
- technical, minimal in reco (the changes in /test/ were not reviewed in detail)
- no changes expected/observed

**Comment by perrotta** [comment]
+1

**Pull request merged**